### PR TITLE
[1.19] kubernetes v1.19.5 with go 1.15.x

### PR DESCRIPTION
**Pull request by dweomer**
Bump kubernetes and related dependencies, including k3s, to v1.19.x
Bump golang to 1.15.x to match upstream

Addresses #485
Addresses #534

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -59,8 +59,8 @@ VOLUME /var/lib/rancher/k3s
 
 FROM build AS build-k8s-codegen
 ARG KUBERNETES_VERSION
-RUN git clone -b ${KUBERNETES_VERSION} --depth=1 https://github.com/kubernetes/kubernetes.git ${GOPATH}/src/github.com/kubernetes/kubernetes
-WORKDIR ${GOPATH}/src/github.com/kubernetes/kubernetes
```
Comment (dweomer) [inline Dockerfile:63]: No need to potentially trick go that it is building an old-school GOPATH project.

File: Dockerfile [hunk 2]
```diff
@@ -85,13 +85,12 @@ RUN echo "export GO_LDFLAGS=\"-linkmode=external \
...
     -X k8s.io/client-go/pkg/version.buildDate=\${BUILD_DATE} \
     \"" >> /usr/local/go/bin/go-build-static-k8s.sh
-RUN echo 'go-build-static.sh -gcflags=-trimpath=${GOPATH}/src/github.com/kubernetes/kubernetes -mod=vendor -tags=selinux,osusergo,netgo ${@}' \
+RUN echo 'go-build-static.sh -gcflags=-trimpath=${GOPATH}/src/kubernetes -mod=vendor -tags=selinux,osusergo,netgo ${@}' \
     >> /usr/local/go/bin/go-build-static-k8s.sh
 RUN chmod -v +x /usr/local/go/bin/go-*.sh
 
 FROM build-k8s-codegen AS build-k8s
 RUN go-build-static-k8s.sh -o bin/kube-apiserver           ./cmd/kube-apiserver
-RUN go-build-static-k8s.sh -o bin/apiextensions-apiserver  ./vendor/k8s.io/apiextensions-apiserver
```
Comment (dweomer) [inline Dockerfile:94]: This wasn't building so I removed it, with prejudice.

File: go.mod [hunk 3]
```diff
@@ -15,51 +15,51 @@ replace (
 	github.com/containerd/typeurl => github.com/containerd/typeurl v0.0.0-20180627222232-a93fcdb778cd
 	github.com/coreos/flannel => github.com/rancher/flannel v0.12.0-k3s1
 	github.com/coreos/go-systemd => github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
-	github.com/docker/distribution => github.com/docker/distribution v0.0.0-20190205005809-0d3efadf0154
-	github.com/docker/docker => github.com/docker/docker v17.12.0-ce-rc1.0.20190219214528-cbe11bdc6da8+incompatible
```
Comment (dweomer) [inline go.mod:19]: These docker import replacements match the k3s 1.19 branch

File: go.mod [hunk 4]
```diff
@@ -15,51 +15,51 @@ replace (
...
+	k8s.io/sample-cli-plugin => github.com/k3s-io/kubernetes/staging/src/k8s.io/sample-cli-plugin v1.19.5-k3s1
+	k8s.io/sample-controller => github.com/k3s-io/kubernetes/staging/src/k8s.io/sample-controller v1.19.5-k3s1
 	mvdan.cc/unparam => mvdan.cc/unparam v0.0.0-20190209190245-fbb59629db34
 )
 
 require (
 	github.com/google/go-containerregistry v0.0.0-20190617215043-876b8855d23c
 	github.com/pkg/errors v0.9.1
 	github.com/rancher/helm-controller v0.7.3
-	github.com/rancher/k3s v1.19.2-0.20201028222356-523ccaf3f22a
```
Comment (dweomer) [inline go.mod:62]: This used to be on k3s master. I double-checked that the referenced tag wasn't missing anything that rke2 might need.

**Comment by dweomer** [comment]
```
bash-5.0# kubectl get node,pod -A -o wide
NAME               STATUS   ROLES         AGE     VERSION                      INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
node/rke2-server   Ready    etcd,master   2m17s   v1.19.5-dev+5236c5b9-dirty   172.17.0.2    <none>        Alpine Linux v3.12   5.4.0-56-generic   containerd://1.3.9-k3s1

NAMESPACE     NAME                                                     READY   STATUS         RESTARTS   AGE    IP           NODE          NOMINATED NODE   READINESS GATES
kube-system   pod/etcd-rke2-server                                     1/1     Running        0          100s   172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-canal-556vm                        0/1     NodeAffinity   0          2m2s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-canal-72pb9                        0/1     NodeAffinity   0          112s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-canal-htltx                        0/1     Completed      0          72s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-canal-pdjkp                        0/1     NodeAffinity   0          2m2s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-coredns-75vnn                      0/1     Completed      0          91s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-coredns-cl5g5                      0/1     NodeAffinity   0          111s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-coredns-k5dsh                      0/1     NodeAffinity   0          2m2s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-coredns-xlvs8                      0/1     NodeAffinity   0          2m1s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-ingress-nginx-n844x                0/1     Completed      0          2m2s   10.42.0.3    rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-kube-proxy-m964v                   0/1     NodeAffinity   0          2m     <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-kube-proxy-pfcmv                   0/1     Completed      0          90s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-kube-proxy-pg26g                   0/1     NodeAffinity   0          2m2s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-kube-proxy-pq9n4                   0/1     NodeAffinity   0          110s   <none>       rke2-server   <none>           <none>
kube-system   pod/helm-install-rke2-metrics-server-hq76t               0/1     Completed      0          2m2s   10.42.0.2    rke2-server   <none>           <none>
kube-system   pod/kube-apiserver-rke2-server                           1/1     Running        0          72s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/kube-controller-manager-rke2-server                  1/1     Running        0          69s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/kube-proxy-hlm8p                                     1/1     Running        0          89s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/kube-scheduler-rke2-server                           1/1     Running        0          47s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/rke2-canal-jpl2s                                     2/2     Running        0          71s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/rke2-coredns-rke2-coredns-6975675896-28b7v           1/1     Running        0          89s    10.42.0.6    rke2-server   <none>           <none>
kube-system   pod/rke2-ingress-nginx-controller-54946dd48f-7d7t5       1/1     Running        0          54s    172.17.0.2   rke2-server   <none>           <none>
kube-system   pod/rke2-ingress-nginx-default-backend-5795954f8-mrbmr   1/1     Running        0          54s    10.42.0.5    rke2-server   <none>           <none>
kube-system   pod/rke2-metrics-server-68f9fb6485-gclrh                 1/1     Running        0          54s    10.42.0.4    rke2-server   <none>           <none>
```

Ignore the `NodeAffinity` statuses, known bug.

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Comment by dweomer** [comment]
These changes were merged as part of #599, so, closing this in favor of that.

**Pull request closed without merging**